### PR TITLE
Modal creation

### PR DESCRIPTION
**Pull request by tomriiich**

### CODE REVIEW

File: src/components/HeroModal.jsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    <div className={show ? 'heroSide' : 'hide'}>
+      <button type="button" onClick={() => setShow(false)}>X</button>
+      {heroList.map((hero) => (
+        <Card
+          key={hero.id}
+          avatar={hero}
+          fighterId={heroFighterId}
+          setFighterId={setHeroFighterId}
+        />
+      ))}
```
Comment (rocambille) [inline src/components/HeroModal.jsx]: use children to fill the modal, and reuse one Modal component for both Hero and Villain. something like that:

https://github.com/WildCodeSchool/reims-js-2103-project2-wild-code-tails/blob/main/src/component/CocktailList/Cocktail.jsx#L17
https://github.com/WildCodeSchool/reims-js-2103-project2-wild-code-tails/blob/main/src/component/CocktailList/Modal.jsx

**Review by rocambille — CHANGES_REQUESTED** [review]

File: src/components/HeroModal.jsx (hunk 1 above)
Comment (tomriiich) [reply src/components/HeroModal.jsx]: Modal changes done

File: src/components/Game.jsx [hunk 2]
```diff
@@ -42,15 +44,13 @@ function Game() {
...
-              setFighterId={setHeroFighterId}
-            />
-          ))}
+          <Modal
+            avatarList={heroList}
+            avatarFighterId={heroFighterId}
+            setAvatarFighterId={setHeroFighterId}
+            show={showHero}
+            setShow={setShowHero}
+          />
```
Comment (rocambille) [inline src/components/Game.jsx]: you should have here:

    <Modal>
      {heroList.map((hero) => (
        <Card
          key={hero.id}
          avatar={hero}
          playerTurn={playerTurn}
          fighterId={heroFighterId}
          setFighterId={setHeroFighterId}
       />
      ))}
    </Modal>

File: src/components/Modal.jsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    <div className={show ? 'hand' : 'hide'}>
+      <button type="button" onClick={() => setShow(false)}>X</button>
+      {avatarList.map((hero) => (
+        <Card
+          key={hero.id}
+          avatar={hero}
+          fighterId={avatarFighterId}
+          setFighterId={setAvatarFighterId}
+        />
+      ))}
```
Comment (rocambille) [inline src/components/Modal.jsx]: you should have here:

    {children}

File: src/components/Game.jsx [hunk 4]
```diff
@@ -116,6 +114,15 @@ function Game() {
           </button>
           )}
         </div>
+        {!showVillain && <button className="drawButton" type="button" onClick={() => setShowVillain(true)}>Show Villain hand</button>}
```
Comment (rocambille) [inline src/components/Game.jsx]: this should be in Modal

File: src/components/Game.jsx [hunk 5]
```diff
@@ -17,6 +17,8 @@ function Game() {
   const [villainFighterId, setVillainFighterId] = useState();
   const [villainFighterStr, setVillainFighterStr] = useState();
   const [villainFighterHp, setVillainFighterHp] = useState();
+  const [showHero, setShowHero] = useState(false);
```
Comment (rocambille) [inline src/components/Game.jsx]: this should be in Modal

File: src/components/HeroModal.jsx [hunk 6]
```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Card from './Card';
+import './Modal.css';
+
+function Modal({
```
Comment (rocambille) [inline src/components/HeroModal.jsx]: this file should be removed

File: src/components/Modal.jsx [hunk 7]
```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Card from './Card';
+import './Modal.css';
+
+function Modal({
+  avatarList, avatarFighterId, setAvatarFighterId, show, setShow,
```
Comment (rocambille) [inline src/components/Modal.jsx]: you should have here:

    children

File: src/components/Modal.jsx [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+      id: PropTypes.string,
+      name: PropTypes.string,
+      powerstats: PropTypes.objectOf(PropTypes.string),
+      image: PropTypes.objectOf(PropTypes.string),
+    }),
+  ).isRequired,
+  avatarFighterId: PropTypes.string.isRequired,
+  setAvatarFighterId: PropTypes.func.isRequired,
+  show: PropTypes.bool.isRequired,
+  setShow: PropTypes.func.isRequired,
```
Comment (rocambille) [inline src/components/Modal.jsx]: you should have here:

    children: PropTypes.element,

File: src/components/VillainModal.jsx [hunk 9]
```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Card from './Card';
+import './Modal.css';
+
+function Modal({
```
Comment (rocambille) [inline src/components/VillainModal.jsx]: this file should be removed

**Review by rocambille — CHANGES_REQUESTED** [review]

**Comment by tomriiich** [comment]
Modal problems fixed. Could you please merge Modal into main ? Thanks

**Pull request merged**